File: 11vaFeriaAcademicayCultural/Basico/Problema2/problema2.java
/*Problema 2 Solo numeros
Propuesta de solucion
autor: Luis Arturo De La Garza Navel
Java*/

import java.util.Scanner;

class Problema2{
  public static void main(String[] args) {
    Scanner entrada = new Scanner(System.in);
    String cadena = entrada.nextLine();
    int acumulador = 0; //inicializamos un acumulador para almacenar el valor de los numeros

    for(int i = 0; i < cadena.length(); i++){
      char elemento = cadena.charAt(i); // accedemos al caracter en el indice i
      if(Character.isDigit(elemento)){// verificamos si es digito con isDigit de Character
        /*convertimos al caracter en numero con el metodo parseInt de la clase Interger
        forzando al caracter a ser un String, dado que el tipo de dato que acepta dicho metodo*/
        acumulador += Integer.parseInt(""+elemento);
      }
    }
    System.out.println(acumulador);
  }
}

File: 11vaFeriaAcademicayCultural/Basico/Problema2/problema2.cpp
/*Problema 2 Solo numeros
Propuesta de solucion
autor: Luis Arturo De La Garza Navel
C++*/

#include <iostream>
#include<string> //libreria para manejo de la clase string
#include<ctype.h> // libreria para la funcion que verifica si un caracter es digito
using namespace std;

int main() {
  string entrada; // string para leer la entrada
  int acumulador = 0; // int para acumular los caracteres que son digitos

  cin>>entrada;

  for(int i = 0; i < entrada.length(); i++){
    if(isdigit(entrada[i])){
      /* le restamos a caracter 48, que es el del cero (primer numero) de Ascii*/
      acumulador += entrada[i] - 48;
    }
  }
  cout << acumulador << endl;

  return 0;
}
